test/webaudio/audioworklet_2x_in_out_stereo.c

```diff
@@ -0,0 +1,156 @@
+  return true;
+}
+
+// Registered click even to (1) enable audio playback and (2) toggle playing the tracks
+bool onClick(int type, const EmscriptenMouseEvent* e, void* data) {
+  EMSCRIPTEN_WEBAUDIO_T ctx = VOIDP_2_WA(data);
+  if (emscripten_audio_context_state(ctx) != AUDIO_CONTEXT_STATE_RUNNING) {
+    printf("Resuming playback\n");
+    emscripten_resume_audio_context_sync(ctx);
+  }
+  printf("Toggling audio playback\n");
+  toggleTrack(beatID);
+    toggleTrack(bassID);
+  return false;
+}
+
+// Audio processor created, now register the audio callback
+void processorCreated(EMSCRIPTEN_WEBAUDIO_T context, bool success, void* data) {
+  if (success) {
+    printf("Audio worklet processor created\n");
+    printf("Click to toggle audio playback\n");
+
+    // Two stereo outputs, two inputs
+    int outputChannelCounts[2] = { 2, 2 };
+    EmscriptenAudioWorkletNodeCreateOptions opts = {
+      .numberOfInputs  = 2,
+      .numberOfOutputs = 2,
+      .outputChannelCounts = outputChannelCounts
+    };
+    EMSCRIPTEN_AUDIO_WORKLET_NODE_T worklet = emscripten_create_wasm_audio_worklet_node(context, "mixer", &opts, &process, NULL);
+    // Both outputs connected to the context
+    emscripten_audio_node_connect(worklet, context, 0, 0);
+    emscripten_audio_node_connect(worklet, context, 1, 0);
+
+    // Create the two stereo source nodes and connect them to the two inputs
+    // Note: we can connect the sources to the same input and it'll get mixed for us, but that's not the point
+    beatID = createTrack(context, "audio_files/emscripten-beat.mp3", true);
+    if (beatID) {
+      emscripten_audio_node_connect(beatID, worklet, 0, 0);
+    }
+    bassID = createTrack(context, "audio_files/emscripten-bass.mp3", true);
+    if (bassID) {
+      emscripten_audio_node_connect(bassID, worklet, 0, 1);
+    }
+
+    // Register a click to start playback
+    emscripten_set_click_callback(EMSCRIPTEN_EVENT_TARGET_DOCUMENT, WA_2_VOIDP(context), false, &onClick);
+
+    // Register the counter that exits the test after one second of mixing
+    emscripten_set_timeout_loop(&playedAndMixed, 16, NULL);
+  } else {
+    printf("Audio worklet node creation failed\n");
+  }
+}
+
+// Worklet thread inited, now create the audio processor
+void initialised(EMSCRIPTEN_WEBAUDIO_T context, bool success, void* data) {
+  if (success) {
+    printf("Audio worklet initialised\n");
+
+    WebAudioWorkletProcessorCreateOptions opts = {
+      .name = "mixer",
+    };
+    emscripten_create_wasm_audio_worklet_processor_async(context, &opts, &processorCreated, NULL);
+  } else {
+    printf("Audio worklet failed to initialise\n");
+  }
+}
+
+int main() {
+  static char workletStack[AUDIO_STACK_SIZE];
+  EMSCRIPTEN_WEBAUDIO_T context = emscripten_create_audio_context(NULL);
+  emscripten_start_wasm_audio_worklet_thread_async(context, workletStack, sizeof workletStack, &initialised, NULL);
+  emscripten_runtime_keepalive_push();
+  return 0;
```

Comment:
There seems to be a lot of duplication between these new tests.. maybe thats fine?  Maybe we could put the shared stuff in the common header?

Reply:
They're pretty much the same common code with tweaks per file, but I wanted to keep them self contained.

I've no objection to splitting them into common code, I just don't think it helps people looking at the file for ideas.

Reply:
As a reviewer I think it might be nice to just review stuff once rather than having keep checking which things are unchanged.     I think maybe a shared header with common code could be good actually.

Reply:
Over half the functions do differ, sometimes by quite a lot:

![Screenshot 2025-01-14 at 16 57 20](https://github.com/user-attachments/assets/04564c64-a4a8-4c28-93c9-51d45f4bfc3f)

I was kinda torn between standalone examples and something easy to maintain (and something quick to write, since the actual performance improvement was the aim).

Reply:
I could split the common unchanged code into its own included file, since I do get your point about reviewing and later changing (I just made the same edit four times). I'm not sure it's worth the tradeoff of the single-file simplicity. But I'm not fussy.

Reply:
> I think maybe a shared header with common code could be good actually.

Done, I split out the code that remains the same for all four tests, which more than halves their size.